test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnVerbs.java

```diff
@@ -330,6 +353,105 @@ public static HapiContractUpdate contractUpdate(String contract) {
+				final var path = getResourcePath(contractName, ".bin");
+				final var file = new HapiFileCreate(contractName);
+				final var updatedFile = updateLargeFile(GENESIS, contractName, extractByteCode(path));
+				ops.add(file);
+				ops.add(updatedFile);
+			}
+			allRunFor(spec, ops);
+		});
+	}
+
+	/**This method enables the developer to upload contract bytecode with the ability to chain methods, related to HapiFileCreate.
+	 The execution of the method requires the op context, therefore the invocation is not straightforward.
+	 Example:
+	 withOpContext(
+	 (spec, noOp) -> {
+	 chainedFileCreate(contract, spec).hasPrecheck(TRANSACTION_OVERSIZE);
+	 chainedFileCreate(contract, spec).contents("").key(KEY_LIST);
+	 }
+	 )
+	 */
+	public static HapiFileCreate uploadInitCode(final String contractName, final HapiApiSpec spec) {
+		final var path = getResourcePath(contractName, ".bin");
+		final var file = new HapiFileCreate(contractName);
+		sourcing(() -> file).execFor(spec);
+		sourcing(() -> updateLargeFile(GENESIS, contractName, extractByteCode(path))).execFor(spec);
+		return file;
+	}
+
+	/*  Note to the reviewer:
+		This method is temporarily named with the "new" prefix, as soon as the implementation is approved and the legacy
+		contractCreate() is entirely replaced, this method will be renamed to "contractCreate"
+	 */
+	public static HapiContractCreate newContractCreate(final String contractName, final Object... constructorParams) {
+		if (constructorParams.length > 0) {
+			final var constructorABI = getABIFor(CONSTRUCTOR, EMPTY, contractName);
+			return new HapiContractCreate(contractName, constructorABI, constructorParams).bytecode(contractName);
+		} else {
+			return new HapiContractCreate(contractName).bytecode(contractName);
+		}
+	}
+
+	/**  Note:
+	 Previously - when creating contracts we were passing a name, chosen by the developer, which can differentiate
+	 from the name of the contract.
+	 The new implementation of the contract creation depends on the exact name of the contract, which means, that we
+	 can not create multiple instances of the contract with the same name.
+	 Therefore - in order to provide each contract with a unique name - the developer must attach a suffix to happily
+	 create multiple instances of the same contract, but with different names.
+	 Example:
+	 final var contract = "TransferringContract";
+	 newContractCreate(contract).balance(10_000L).payingWith(ACCOUNT),
+	 newContractCreate(contract, to).balance(10_000L).payingWith(ACCOUNT)
+	 */
+	public static HapiContractCreate cloneContract(final String contractName, final String suffix, final Object... constructorParams) {
+		if (constructorParams.length > 0) {
+			final var constructorABI = getABIFor(CONSTRUCTOR, EMPTY, contractName);
+			return new HapiContractCreate(contractName + suffix, constructorABI, constructorParams).bytecode(contractName);
+		} else {
+			return new HapiContractCreate(contractName + suffix).bytecode(contractName);
+		}
+	}
```

Comment:
I don't think adding this `cloneContract` method is a good idea... generally we use the TxnVerbs class as a mirror to what the HAPI API provides as interface. There is no `clone` transaction in the HAPI API and therefore this is confusing. 
Instead of requiring the E2E devs to know and use `cloneContract` I suggest adding one more `contractCreate` method with the following params:
```
	public static HapiContractCreate newContractCreate(final String contractTestName, final String bytecodeFileName, final Object... constructorParams) 
```
Then make the already existing `newContractCreate` method call this one like this:
```
	public static HapiContractCreate newContractCreate(final String bytecodeFileName,
													   final Object... constructorParams) {
		return newContractCreate(bytecodeFileName, bytecodeFileName, constructorParams);
	}
```

Reply:
After a discussion with the reviewer - this implementation remains intact, but with a different, less confusing name of the method - contractCustomCreate().